tests/unattended/unit/framework/bach.sh

```diff
@@ -0,0 +1,642 @@
+    eval "function @${donotpanic}() { builtin printf '\n%s\n  line number: %s\n  script stack: %s\n\n' 'DO NOT PANIC!' \"\${BASH_LINENO}\" \"\${BASH_SOURCE[*]}\"; builtin exit 1; } >&2; export -f @${donotpanic};"
+done
+
+function bach--is-function() {
+    [[ "$(@type -t "$1")" == function ]]
+}
+export -f bach--is-function
+
+declare -gr __bach_run_test__ignore_prefix="## BACH:"
+function @comment() {
+    @out "${__bach_run_test__ignore_prefix}" "$@"
+}
+export -f @comment
+
+function bach-run-tests() {
+    set -euo pipefail
+
+    bach_initialize
+
+    for donotpanic in donotpanic dontpanic do-not-panic dont-panic do_not_panic dont_panic; do
+        eval "function @${donotpanic}() { builtin true; }; export -f @${donotpanic}"
+    done
+
+    function command() {
+        if [[ "$1" != -* ]] && bach--is-function "$1"; then
+            "$@"
+        else
+            mockfunc="$(@generate_mock_function_name command "$@")"
+            if bach--is-function "${mockfunc}"; then
+                @debug "[BC-func]" "${mockfunc}" "$@"
+                "${mockfunc}" "$@"
+            else
+                command_not_found_handle command "$@"
+            fi
+        fi
+    }
+    export -f command
+
+    function xargs() {
+        declare param
+        declare -a xargs_opts
+        while param="${1:-}"; [[ -n "${param:-}" ]]; do
+            shift || true
+            if [[ "$param" == "--" ]]; then
+                xargs_opts+=("${BASH:-bash}" "-c" "$(builtin printf "'%s' " "$@") \$@" "-s")
+                break
+            else
+                xargs_opts+=("$param")
+            fi
+        done
+        @debug "@mock-xargs" "${xargs_opts[@]}"
+        if [[ "$#" -gt 0 ]]; then
+            @xargs "${xargs_opts[@]}"
+        else
+            [[ -t 0 ]] || @cat &>/dev/null
+            @dryrun xargs "${xargs_opts[@]}"
+        fi
+    }
+    export -f xargs
+
+    if [[ "${BACH_ASSERT_IGNORE_COMMENT}" == true ]]; then
+        BACH_ASSERT_DIFF_OPTS+=(-I "^${__bach_run_test__ignore_prefix}")
+    fi
+
+    declare color_ok color_err color_end
+    if [[ "$BACH_COLOR" == "always" ]] || [[ "$BACH_COLOR" != "no" && -t 1 && -t 2 ]]; then
+        color_ok="\e[1;32m"
+        color_err="\e[1;31m"
+        color_end="\e[0;m"
+    else
+        color_ok=""
+        color_err=""
+        color_end=""
+    fi
+    declare name friendly_name testresult test_name_assert_fail
+    declare -i total=0 error=0
+    declare -a all_tests
+    mapfile -t all_tests < <(bach-run-tests--get-all-tests)
+    @echo "1..${#all_tests[@]}"
+    for name in "${all_tests[@]}"; do
+        # @debug "Running test: $name"
+        friendly_name="${name/#test-/}"
+        friendly_name="${friendly_name//-/ }"
+        friendly_name="${friendly_name//  / -}"
+        : $(( ++total ))
+        testresult="$(@mktemp)"
+        @set +e
+        assert-execution "$name" &>"$testresult"; test_retval="$?"
```

Comment:
Same

Reply:
`bach.sh` is the external framework. We will not modify it.